analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-table.Rmd

```diff
@@ -0,0 +1,529 @@
+                                "sample_id")) %>%
+  rename(Kids_First_Biospecimen_ID_DNA = Kids_First_Biospecimen_ID.x,
+         Kids_First_Biospecimen_ID_RNA = Kids_First_Biospecimen_ID.y) %>%
+  select(starts_with("Kids_First"), sample_id, everything())
+
+head(all_data_df, n = 10)
+```
+
+Add in brain region and age.
+
+```{r}
+relevant_clinical_df <- histologies_df %>%
+  filter(sample_id %in% all_data_df$sample_id,
+         sample_type == "Tumor",
+         composition == "Solid Tissue") %>%
+  mutate(age_at_diagnosis_yr = 
+           floor(as.integer(age_at_diagnosis_days)/365)) %>%
+  group_by(Kids_First_Participant_ID, sample_id) %>%
+  summarize(glioma_brain_region = paste(sort(unique(glioma_brain_region)), 
+                                        collapse = ", "),
+            age_at_diagnosis_yr = paste(sort(unique(age_at_diagnosis_yr)),
+                                        collapse = ", "))
+```
+
+```{r}
+all_data_df <- inner_join(relevant_clinical_df,
+                          all_data_df)
+```
+
+Write to file.
+
+```{r}
+write_tsv(all_data_df, output_all)
+```
+
+## Tables by Molecular Subtype
+
+In this section, we'll divide up the table based on the molecular subtypes described in [`AlexsLemonade/OpenPBTA-analysis#249`](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/249).
+
+### H3 K28 mutant
+
+> * These tumors contain H3F3A K28M or HIST1H3B K28M mutations
+> * Co-occurring lesions include: ACVR1, TP53, ATRX mutations; PDGFRA amplification; PTEN loss
+> * Mutually-exclusive lesions: FGFR1 mutations/fusions (thalamic); IDH1 mutations; BRAF V600E (low-grade gliomas)
+> * Average age of 9 years
+> * Majority should be midline localized
+
+```{r}
+h3_k28_df <- all_data_df %>% 
+  # Only rows with H3 K28M mutations
+  filter(H3F3A.K28M == "Yes" | HIST1H3B.K28M == "Yes") %>%
+  # Only the relevant columns
+  select(Kids_First_Participant_ID, 
+         sample_id,
+         H3F3A.K28M,
+         HIST1H3B.K28M,
+         H3F3A.G35R,
+         H3F3A.G35V,
+         age_at_diagnosis_yr,
+         glioma_brain_region,
+         relevant_coding_mutations,
+         PTEN_focal_status,
+         PDGFRA_focal_status,
+         BRAF_V600E,
+         FGFR1_fusions) %>%
+  distinct()
+
+h3_k28_df
+```
+
+Mutually exclusive with the H3 G35 mutations?
+
+```{r}
+any(h3_k28_df$H3F3A.G35R == "Yes")
+```
+
+```{r}
+any(h3_k28_df$H3F3A.G35V == "Yes")
+```
+
+Are the majority midline localized?
+
+```{r}
+table(h3_k28_df$glioma_brain_region)
+```
+
+Yes.
+
+```{r}
+h3_k28_df %>%
+  group_by(relevant_coding_mutations) %>%
+  count()
+```
+
+The coding mutation list looks like what we would expect based on the description, with the exception of some BRAF mutations.
+There are no _IDH1_ mutations.
+
+```{r}
+h3_k28_df %>%
+  group_by(PTEN_focal_status, PDGFRA_focal_status) %>%
+  count()
+```
+
+The copy number information looks a bit different than what is described, but we are not yet using consensus calls.
```

Comment:
This looks fine - when I wrote in co-occurring, those are just _**some**_ of the lesions that had been reported as co-occurring, as opposed to lesions that must/usually co-occur.